server/controllers/image.js

```diff
@@ -0,0 +1,113 @@
+const aws = require("aws-sdk");
+const express = require("express");
+const router = express.Router();
+const fileUpload = require("express-fileupload");
+require("dotenv").config();
+
+router.use(fileUpload());
+
+// AWS S3 configuration
+const bucketName = process.env.S3_BUCKET_NAME;
+aws.config.credentials = new aws.Credentials({
+  accessKeyId: process.env.S3_ACCESS_KEY,
+  secretAccessKey: process.env.S3_SECRET_ACCESS_KEY
+});
+
+// Get all folder from bucket
+const listObjects = (bucket) => {
+  const listObjects = new Promise((resolve, reject) => {
+    bucket.listObjects((error, data) => {
+      error ? console.log(error) : resolve(data.Contents);
+    });
+  });
+  return listObjects;
+};
+
+// Get all file from folder
+const getSingleObject = (bucket, key) => {
+  const getSingleObject = new Promise((resolve, reject) => {
+    bucket.getObject({ Bucket: bucketName, Key: key }, (error, data) => {
+      error ? console.log(error) : resolve(data.Body.toString("base64"));
+    });
+  });
+  return getSingleObject;
+};
+
+// Save images inside folder
+const saveObject = (bucket, params) => {
+  const saveObject = new Promise((resolve, reject) => {
+    bucket.putObject(params, (error, data) => {
+      error ? console.error("error: ", error) : resolve(data);
+    });
+  });
+  return saveObject;
+}
+
+// Delete folder
+const deleteObjects = (bucket, objects) => {
+  const deleteObjects = new Promise((resolve, reject) => {
+    bucket.deleteObjects({ Delete: { Objects: objects, Quiet: true }}, (error, data) => {
+      error ? console.error("error: ", error) : resolve(data);
+    });
+  });
+  return deleteObjects;
+}
+
+// Get images by business id
+router.get("/:id", async (req, res) => {
+  const folderName = req.params.id;
```

Comment:
I believe we are going to fetch images in two places:
- business profile => `get 'api/users/account'` 
- list/map view => `get 'api/businesses/` + location, date, time, queries 

Should we just include this logic in those endpoints? What do you think?

Reply:
@catboop Oh yeah, that's right...and you mean `list/map view` is `api/availability` ? And so, are we using by email not business id? Because there is no email when getting availability so I'll need to add a bit 👍

Reply:
Oops yes, `api/availability`! For that one, we are just fetching businesses by `location`, `date`, start_hour`, and `end_hour`. 

For `api/users/account`, we fetch by user's email. 

Does it make sense to add this logic to both endpoints so that `images` are just part of the responses?

Reply:
@catboop That could work but I was thinking that it might be little laggy to do multiple things in one endpoint. (I think just image itself will take bit time). And just to be clear what the endpoints are doing, I just made it separately. Is fetching multiple calls not efficient in frontend or mendokusai to manage states(context)?

Reply:
let me give it a try and see how it goes! i was under the impression that it would be more laggy to have multiple fetch calls from the frontend rather than on the backend, but that was just an assumption.